modules/021-cni-cilium/images/cilium/patches/001-netfilter-compatibility-mode.patch

```diff
@@ -1,314 +0,0 @@
---- a/daemon/cmd/daemon_main.go	(revision b0d3140aa079b31d2f3889b0f695cf7add393e68)
-+++ b/daemon/cmd/daemon_main.go	(date 1653900639977)
-@@ -1026,6 +1026,15 @@
- 	flags.Bool(option.EnableCustomCallsName, false, "Enable tail call hooks for custom eBPF programs")
- 	option.BindEnv(option.EnableCustomCallsName)
-
-+	flags.Bool(option.NetfilterCompatibleMode, false, "If set to true, guarantees that traffic will pass through netfilter and that iptable rules will be enforced. This mode may reduce network throughput. If set to false (default), it does not guarantee that all traffic will pass through netfilter. This option only affects NodePort traffic.")
-+	// Currently, NetfilterCompatibleMode guarantees only the nodeport traffic can pass through
-+	// netfilter. Once it guarantees all traffic can pass through netfilter,
-+	// this flag NetfilterCompatibleMode can be marked as visible to users.
-+	// below GH issue tracks list of issues to be fixed for to guarantee all traffic pass through netfilter
-+	// https://github.com/cilium/cilium/issues/16166
-+	flags.MarkHidden(option.NetfilterCompatibleMode)
-+	option.BindEnv(option.NetfilterCompatibleMode)
-+
- 	flags.Bool(option.BGPAnnounceLBIP, false, "Announces service IPs of type LoadBalancer via BGP")
- 	option.BindEnv(option.BGPAnnounceLBIP)
-
-@@ -1554,6 +1563,19 @@
- 			)
- 		}
- 	}
-+
-+	// Enabling NetfilterCompatibleMode increases BPF instruction count size, and may
-+	// cause issues in kernels that have lower instruction complexity limit. Hence enabling
-+	// this feature only in kernels with a higher instruction complexity limit (5.2 or later).
-+	// NetfilterCompatibleMode feature can be enabled on all kernels, once we have support
-+	// for adding new tail calls in host device[bpf_host].
-+	// If kernel doesn't support higher instruction complexity limit, then disable NetfilterCompatibleMode.
-+	if option.Config.NetfilterCompatibleMode {
-+		if !probes.NewProbeManager().GetMisc().HaveLargeInsnLimit {
-+			option.Config.NetfilterCompatibleMode = false
-+			log.Warn("netfilter-compatible-mode requires kernel version 5.2 or higher.")
-+		}
-+	}
- }
-
- func (d *Daemon) initKVStore() {
-Index: bpf/bpf_host.c
-IDEA additional info:
-Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
-<+>UTF-8
-===================================================================
-diff --git a/bpf/bpf_host.c b/bpf/bpf_host.c
---- a/bpf/bpf_host.c	(revision b0d3140aa079b31d2f3889b0f695cf7add393e68)
-+++ b/bpf/bpf_host.c	(date 1653475554160)
-@@ -1081,6 +1081,19 @@
- 	 (defined(ENABLE_DSR) && defined(ENABLE_DSR_HYBRID)) || \
- 	 defined(ENABLE_MASQUERADE) || \
- 	 defined(ENABLE_EGRESS_GATEWAY))
-+
-+#ifdef NETFILTER_COMPAT_MODE
-+
-+	/* For a packet which is a reply from a local endpoint to a NodePort request,
-+	 * do rev-DNAT. To determine if it is Nodeport traffic, do conntrack lookup for
-+	 * all reply packets.
-+	 */
-+	ret = rev_nodeport_lb(ctx);
```

Comment:
Please, verify that NodePort Services that point to `hostNetwork: true` Pods work with DSR without this patch.

Reply:
Is there a clear scenario for how this can be checked and what the result should be?

Reply:
1. Set up a cluster and set DSR mode in cni-cilium config
2. Create a type: LoadBalancer or NodePort Service that points to `hostNetwork: true` Pod and its trafficPolicy set to `Cluster`.
3. Perform a request to the Node without the Pod so that it gets DSR'ed to the other Node.
4. Check whether request succeeds